src/views/CountryRiskWatch/ReturnPeriodTable/index.tsx

```diff
@@ -0,0 +1,199 @@
+    return option.frequencyDisplay;
+}
+
+interface Props {
+    data: CountryRiskResponse[number]['return_period_data'] | undefined;
+}
+
+function ReturnPeriodTable(props: Props) {
+    const { data } = props;
+    const strings = useTranslation(i18n);
+
+    const [hazardType, setHazardType] = useInputState<HazardType>('FL');
+    const hazardOptions = useMemo(
+        () => (
+            unique(
+                data?.map(
+                    (datum) => {
+                        if (isFalsyString(datum.hazard_type) || datum.hazard_type === '') {
+                            return undefined;
+                        }
+
+                        return {
+                            hazard_type: datum.hazard_type,
+                            hazard_type_display: datum.hazard_type_display,
+                        };
+                    },
+                ).filter(isDefined) ?? [],
+                (datum) => datum.hazard_type_display,
+            )
+        ),
+        [data],
+    );
+
+    const columns = [
+        createStringColumn<TransformedReturnPeriodData, string | number>(
+            'frequency',
+            strings.returnPeriodTableReturnPeriodTitle,
+            (item) => item.frequencyDisplay,
+        ),
+        createNumberColumn<TransformedReturnPeriodData, string | number>(
+            'numRiskOfDisplacement',
+            strings.returnPeriodTableReturnPeriodTitle,
+            (item) => item.displacement,
+            {
+                headerInfoTitle: strings.returnPeriodTableDisplacementTitle,
+                headerInfoDescription: resolveToComponent(
+                    strings.returnPeriodTableDisplacementDescription,
+                    {
+                        source: (
+                            <Link
+                                to="https://www.internal-displacement.org/database/global-displacement-risk-model"
+                            >
+                                {strings.returnPeriodTableDisplacementSource}
+                            </Link>
+                        ),
+                    },
+                ),
+                maximumFractionDigits: 0,
+            },
+        ),
+        createNumberColumn<TransformedReturnPeriodData, string | number>(
+            'economicLosses',
+            strings.returnPeriodTableEconomicLossesTitle,
+            (item) => item.economicLosses,
+            {
+                headerInfoTitle: strings.returnPeriodTableEconomicLossesTitle,
+                headerInfoDescription: resolveToComponent(
+                    strings.returnPeriodTableEconomicLossesDescription,
+                    {
+                        source: (
+                            <Link to="https://www.gfdrr.org/en/disaster-risk-country-profiles">
+                                {strings.returnPeriodTableEconomicLossesSource}
+                            </Link>
+                        ),
+                    },
+                ),
```

Comment:
Does this not have any performance impact?

Reply:
Lets benchmark it